### ISSUE TITLE
Add boot testing to PRs

### ISSUE THREAD

**Issue by mwasilew** [qualcomm-linux/meta-qcom-hwe#105]
Boot testing was added in PR #96. This allows to boot test in nightly builds and pushes. There is a mechanism to allow for boot testing on PRs: https://securitylab.github.com/resources/github-actions-preventing-pwn-requests/. Based on this description it should be possible to add boot testing on PRs using `workflow_run` trigger.

**Comment by ndechesne** [comment]
Yes, this is correct, pull_request_target would allow us to do that. That's how we did it for meta-qcom in Linaro CI (https://github.com/Linaro/meta-qcom/blob/master/.github/workflows/premerge.yml).

**Comment by quic-kdybcio** [comment]
Make sure to only run these tests for approved runs, so that no one gets the funny idea of setting some voltage regulators too high on our CI farm ;)

**Comment by mwasilew** [comment]
I think physical damage to the hardware is least of my security concerns. Right now the tests are only scheduled for merges and nightly builds.
